api/global/handler.go

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package global
+
+import (
+	"log"
+	"os"
+	"sync"
+
+	"go.opentelemetry.io/otel/api/oterror"
+)
+
+var defaultHandler = &handler{
+	l: log.New(os.Stderr, "", log.LstdFlags),
+}
+
+// Global Handler (used for testing).
+var globalHandler oterror.Handler = defaultHandler
+
+// handler logs all errors to STDERR.
+type handler struct {
+	sync.Mutex
+	delegate oterror.Handler
+
+	l *log.Logger
+}
+
+func (h *handler) setDelegate(d oterror.Handler) {
+	h.Lock()
+	defer h.Unlock()
+	if h.delegate != nil {
+		// delegate already registered
+		return
+	}
+
+	h.delegate = d
+}
+
+// Handle implements oterror.Handler.
+func (h *handler) Handle(err error) {
+	h.Lock()
+	defer h.Unlock()
+	if h.delegate != nil {
```

Comment:
I'm slightly concerned about deadlock. Maybe move the call to `delegate.Handle()` outside the lock?